lib/rules/selector-attribute-brackets-space-inside/index.js

```diff
@@ -143,49 +149,53 @@ function rule(expectation, options, context) {
 					},
 			  };
 
-		if (expectation === 'always') {
+		if (primary === 'always') {
 			setAttrBefore(attrBefore.replace(/^\s*/, ' '));
-		} else if (expectation === 'never') {
+		} else if (primary === 'never') {
 			setAttrBefore(attrBefore.replace(/^\s*/, ''));
 		}
 	}
 
+	/**
+	 * @param {import('postcss-selector-parser').Attribute} attributeNode
+	 */
 	function fixAfter(attributeNode) {
-		let key;
+		const key = attributeNode.operator
+			? attributeNode.insensitive
+				? 'insensitive'
+				: 'value'
+			: 'attribute';
 
-		if (attributeNode.operator) {
-			key = attributeNode.insensitive ? 'insensitive' : 'value';
-		} else {
-			key = 'attribute';
-		}
+		const rawSpaces = attributeNode.raws.spaces && attributeNode.raws.spaces[key];
+		const rawAfter = rawSpaces && rawSpaces.after;
+
+		const spaces = attributeNode.spaces[key];
 
-		const rawAfter =
-			attributeNode.raws.spaces &&
-			attributeNode.raws.spaces[key] &&
-			attributeNode.raws.spaces[key].after;
+		/** @type {{ after: string, setAfter: (fixed: string) => void }} */
 		const { after, setAfter } = rawAfter
 			? {
 					after: rawAfter,
 					setAfter(fixed) {
-						attributeNode.raws.spaces[key].after = fixed;
+						rawSpaces.after = fixed;
 					},
 			  }
 			: {
-					after: (attributeNode.spaces[key] && attributeNode.spaces[key].after) || '',
+					after: (spaces && spaces.after) || '',
 					setAfter(fixed) {
 						if (!attributeNode.spaces[key]) attributeNode.spaces[key] = {};
 
+						// @ts-expect-error -- TS2532: Object is possibly 'undefined'.
 						attributeNode.spaces[key].after = fixed;
```

Comment:
[note] Unfortunately, the current TypeScript's compiler cannot detect that `attributeNode.spaces[key]` is present.